cmd/controller-manager/app/controller-manager.go

```diff
@@ -117,6 +119,25 @@ func Run(opts *options.Options) error {
 		klog.Fatalf("Invalid Feature Gate: %v", err)
 	}
 
+	// validate if a federation namespace is configured
+	if len(opts.Config.FederationNamespace) == 0 {
+		klog.Fatalf("The federation namespace must be specified via --federation-namespace")
+	}
+
+	if len(masterURL) == 0 && len(kubeconfig) == 0 && len(opts.Config.FederationNamespace) == 0 {
+		// for in-cluster deployment set the namespace associated
+		// with the service account token
+		data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+		if err != nil {
+			klog.Fatalf("An error occurred while attempting to discover the federation namespace from the service account: %v", err)
+		} else {
+			if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
```

Comment:
There is no need for this conditional or intermediate variable.  The following will suffice:

```golang
opts.Config.FederationNamespace = strings.TrimSpace(string(data))
```

An empty string value is fine.  It won't change anything because a condition of this block is that the federation namespace is not set.

Reply:
Yeah, I was concerned about the empty string here but as you mentioned it won't affect the overall process for setting the federation namespace.